src/langsmith/agent-server-scale.mdx

```diff
@@ -0,0 +1,361 @@
+
+There is no upper limit to N_JOBS_PER_WORKER. However, Queue workers are greedy when fetching new runs, meaning they will try to pick up as many runs they have available jobs and begin executing them right away. If N_JOBS_PER_WORKER is too high and traffic is burstly, you may see uneven worker utilization and potentially longer run execution times.
+
+#### Avoid synchronous blocking operations
+
+Avoid synchronous blocking operations in your code. If required, set [BG_JOB_ISOLATED_LOOPS](/langsmith/env-var#bg-job-isolated-loops) to True to execute each run in a separate event loop and avoid blocking the main event loop.
+
+#### Minimize redundant checkpointing
+
+Minimize redundant checkpointing by setting [`durability`](/oss/langgraph/durable-execution#durability-modes) to the minimum value necessary to ensure your data is durable.
+
+#### Self-hosted
+
+<Note>
+These settings are only required for self-hosted deployments. By default, Cloud deployments already have these best practices enabled.
+</Note>
+
+###### Enable the use of queue workers
+
+By default, the api server manages the queue and does not use queue workers. You can enable the use of queue workers by setting the `queue.enabled` configuration to `true`.
+
+```yaml
+queue:
+  enabled: true
+```
+
+This will allow the api server to offload the queue management to the queue workers, significantly reducing the load on the api server and allowing it to focus on handling requests.
+
+##### Support a number of jobs equal to expected throughput
+
+The more runs you execute in parallel, the more jobs you will need to handle the load. There are two main parameters to scale the available jobs:
+
+- number_of_queue_workers: The number of queue workers provisioned.
+- N_JOBS_PER_WORKER: The number of runs that can be executed at a time by a single queue worker. Defaults to 10.
+
+You can calculate the available jobs with the following equation:
+```
+available_jobs = number_of_queue_workers * N_JOBS_PER_WORKER
+```
+
+Throughput is then the number of runs that can be executed per second by the available jobs:
+```
+throughput_per_second = available_jobs / average_run_execution_time_seconds
+```
+
+So the minimum number of queue workers you should provision to support your expected steady state throughput is:
+```
+number_of_queue_workers = throughput_per_second * average_run_execution_time_seconds / N_JOBS_PER_WORKER
+```
+
+##### Configure autoscaling for bursty workloads
+
+Autoscaling is disabled by default, but should be configured for bursty workloads. Using the same calculations as above, you can determine the maximum number of queue workers you should allow the autoscaler to scale to based on maximum expected throughput.
+
+## Scaling for read load
+
+Read load is primarily driven by the following factors:
+
+- Getting the results of a run
+- Getting the state of a thread
+- Searching for runs, threads, cron jobs and assistants
+- Retrieving checkpoints and long term memory
```

Comment:
Could you add the same set of links here, in case a user lands in this section rather than going from the top of the page to the bottom?